padle/padle.py

```diff
@@ -252,20 +253,18 @@ async def stats(self, ctx):
                 correct_id = await self.config.padle_today()
             else:
                 correct_id = save_daily[day - 1][0]
-            if str(day) in all_guesses and len(all_guesses[str(day)]) != 0 and all_guesses[str(day)][-1] == correct_id:
+            if str(day) in all_guesses and correct_id in all_guesses[str(day)]:
                 cur_streak += 1
                 wins += 1
-            else:
-                if cur_streak > max_streak:
-                    max_streak = cur_streak
-                if day != await self.config.num_days():
-                    cur_streak = 0
-
-        if cur_streak > max_streak:
-            max_streak = cur_streak
+            elif day != await self.config.num_days() or await self.config.user(ctx.author).done():
+                cur_streak = 0
+            if cur_streak > max_streak:
+                max_streak = cur_streak
         all_monsters_guessed = []
         for key, value in all_guesses.items():
             all_monsters_guessed.extend(value)
+        if not all_monsters_guessed:
+            all_monsters_guessed.append(self.FALLBACK_PADLE_MONSTER)
         mode = max(set(all_monsters_guessed), key=all_monsters_guessed.count)
```

Comment:
This is
```py
from collections import Counter
import itertools
flatten = itertools.chain.from_iterable
...

    all_monsters_guessed = flatten(all_guesses.values())  # NOT listified, because we only need it once.
    guess_counts = Counter(all_monsters_guessed)
    mode = max(guess_counts,
                key=guess_counts.get,
                default=self.FALLBACK_PADLE_MONSTER)
```

The last line could also use Counter's `most_common`:
```py
if guess_counts:
    mode, most = guess_counts.most_common(1)[0]
```
but I don't know if that's any better.